docker_services_cli/env.py

```diff
@@ -63,22 +61,75 @@ def _load_or_set_env(services_version, default_version):
     else:
         click.secho(
             f"Environment variable for version {version_from_env} not set \
-            or set to a non-compliant format (dot separated numbers).", fg="red"
+            or set to a non-compliant format (dot separated numbers).",
+            fg="red",
         )
         sys.exit(1)
 
 
+def override_default_env(services_to_override=None):
+    """Override default environment according to list of services with version.
+
+    :param services_to_override: List of service name strings including
+        service version without any separator e.g. ``postgresql11``.
+    """
+    services_to_override = set(services_to_override or []).difference(SERVICES.keys())
+    if services_to_override:
+        num_services_to_override = len(services_to_override)
+        for service_name in SERVICES:
+            for service_override in services_to_override:
+                if service_name in service_override:
+                    service_override_version = service_override.replace(
+                        service_name, ""
+                    )
+                    env_var_with_version = (
+                        f"{service_name.upper()}_{service_override_version}_LATEST"
+                    )
+                    if SERVICES_ALL_DEFAULT_MINOR_VERSIONS.get(env_var_with_version):
+                        os.environ[
+                            f"{service_name.upper()}_VERSION"
+                        ] = SERVICES_ALL_DEFAULT_MINOR_VERSIONS.get(
+                            env_var_with_version
+                        )
+                    else:
+                        click.secho(
+                            f"There is no version {service_override_version} for service {service_name}, using default.",
+                            fg="yellow",
+                        )
```

Comment:
👨‍💻 Changing this to fail instead of just showing a warning. Doing things like `docker-services-cli up postgres111` with the current code would result in running tests with version 9 just because of a typo...